src/atlannot/evaluation.py

```diff
@@ -0,0 +1,221 @@
+    return results
+
+
+def compute_iou(vol_true, vol_pred, region_ids=None):
+    """Compute the intersection over union of given region IDs.
+
+    Parameters
+    ----------
+    vol_true: np.ndarray
+        Input volume considered as ground truth.
+    vol_pred: np.ndarray
+        Input volume considered as prediction.
+    region_ids: list[int] | None
+        List of region IDs to compute the intersection over union.
+        If None, the IoU is computed for all the region IDs present in the volume.
+
+    Returns
+    -------
+    results: dict[int, float]
+        Dictionary containing the region id as keys and the average of the
+        intersection over union of that given region id as values.
+    """
+    results = {}
+    if region_ids is None:
+        labels = np.unique(vol_true)
+        for label in labels:
+            results[label] = iou_score(vol_true, vol_pred, k=label)[0]
+        return results
+
+    for region_id in region_ids:
+        results[region_id] = iou_score(vol_true, vol_pred, k=region_id)[0]
+
+    return results
+
+
+def compute_region_entropy(nissl, atlas, label_value, bins=100):
+    """Compute the entropy of densities of Nissl for a given label value in the atlas.
+
+    Parameters
+    ----------
+    nissl: np.ndarray
+        Nissl volume.
+    atlas: np.ndarray
+        Annotation atlas.
+    label_value: int
+        Label Value to consider.
+    bins: int
+        Number of bins used to compute the histogram of the densities.
+
+    Returns
+    -------
+    region_entropy: float
+        Entropy of the densities of Nissl at a given region value.
+    """
+    prob_region, _ = np.histogram(nissl[atlas == label_value], bins=bins)
+    region_entropy = entropy(prob_region)
+    return region_entropy
+
+
+def compute_entropies(nissl, atlas):
+    """Compute entropies of Nissl densities.
+
+    Parameters
+    ----------
+    nissl: np.ndarray
+        Nissl volume.
+    atlas: np.ndarray
+        Annotation atlas.
+
+    Returns
+    -------
+    brain_entropy: float
+        Entropy of the densities of Nissl for the entire brain.
+
+    conditional_entropy: float
+        Conditional entropy of the densities of Nissl depending on the brain regions.
+    """
+    # 1: For the entire brain
+    prob_entire_brain, _ = np.histogram(nissl[atlas != 0], bins=100)
+    brain_entropy = entropy(prob_entire_brain)
+
+    # 2. Computation of the conditional entropy
+    n_pixels = (atlas != 0).sum()
+    label_values, count_values = np.unique(atlas, return_counts=True)
+    all_region_entropy = []
+    for label, count in zip(label_values, count_values):
+        all_region_entropy.append(compute_region_entropy(nissl, atlas, label) * count)
+
+    conditional_entropy = np.sum(all_region_entropy) / n_pixels
+    return brain_entropy, conditional_entropy
+
+
+def evaluate(
+    atlas,
+    nissl,
+    reference,
+    region_meta,
+    regions_to_evaluate=REGIONS_TO_EVALUATE,
```

Comment:
A mutable default is not a good practise. I'd set it to `regions_to_evaluate=None` and then check for `None` in the function body.

Reply:
Done in 353f218.